base/main.cpp

```diff
@@ -314,6 +314,11 @@ static Common::Error runGame(const Plugin *plugin, const Plugin *enginePlugin, O
 	system.getEventManager()->purgeKeyboardEvents();
 	system.getEventManager()->purgeMouseEvents();
 
+	#if defined(UNCACHED_PLUGINS) && defined(DYNAMIC_MODULES) && !defined(DETECTION_STATIC)
+	PluginManager::instance().unloadDetectionPlugin();
```

Comment:
this could probably be moved higher in the function, hopefully before the engine is allocated, as long as you copy the data that's needed first

Reply:
You can't call unloadDetectionPlugin like this as the Engine may require its MetaEngine to work.